### PR TITLE
Termination Truncation for MPE Environments

### PR DESCRIPTION
**Pull request by jjshoots**
# Description

This updates the termination and truncation API for MPE environments.

### CODE REVIEW

**Comment by jjshoots** [comment]
Hey @WillDudley, do you mind reviewing this?

File: pettingzoo/sisl/waterworld/waterworld.py [hunk 1]
```diff
@@ -71,7 +72,7 @@ def render(self, mode="human"):
         return self.env.render(mode)
 
     def step(self, action):
-        if self.dones[self.agent_selection]:
+        if self.terminations[self.agent_selection]:
```
Comment (WillDudley) [inline pettingzoo/sisl/waterworld/waterworld.py:75]: `if terminations or truncations`?

**Review by WillDudley — COMMENTED** [review]
Looks clean! I have two points I mentioned that could do with clearing up :)

Also, regarding tests:
 Pursuit doesn't pass tests, unsure if this is related to these changes. The error relates to 
```
>       tpos += self.motion_range[a]
E       TypeError: list indices must be integers or slices, not NoneType
```
in discrete_agent.py

Do you know why waterwold's tests are commented out?

File: pettingzoo/sisl/multiwalker/multiwalker.py [hunk 2]
```diff
@@ -76,38 +77,41 @@ def observe(self, agent):
...
+                    iter_agents.remove(agent)
             self._agent_selector.reinit(iter_agents)
         else:
             self._clear_rewards()
         if self._agent_selector.agent_order:
             self.agent_selection = self._agent_selector.next()
 
         if self.env.frames >= self.env.max_cycles:
-            self.dones = dict(zip(self.agents, [True for _ in self.agents]))
+            self.terminations = dict(zip(self.agents, [True for _ in self.agents]))
```
[the remark this answers is not in the document]
Comment (jjshoots) [reply pettingzoo/sisl/multiwalker/multiwalker.py:112]: You're right, also thanks for running tests! I should probably have done it on my local. I'll go have a poke around and try to fix those tests.

File: pettingzoo/sisl/waterworld/waterworld.py (hunk 1 above)
Comment (jjshoots) [reply pettingzoo/sisl/waterworld/waterworld.py:75]: I wanted to do terminations or truncations, but this method allows the user to continue stepping the environment even when truncation is true. This is useful for debugging.

Comment (jjshoots) [reply pettingzoo/sisl/waterworld/waterworld.py:75]: Turns out tests fail when this happens, so I'm reverting to `terminations or truncations`.

**Pull request merged**